recipe/thirdparty/utils.go

```diff
@@ -102,3 +104,107 @@ func parseUsers(value interface{}) ([]tpmodels.User, error) {
 	}
 	return user, nil
 }
+
+func findAndCreateProviderInstance(providers []tpmodels.ProviderInput, thirdPartyId string, tenantId *string) (tpmodels.TypeProvider, error) {
+	for _, provider := range providers {
+		if provider.Config.ThirdPartyId == thirdPartyId {
+			providerInstance := createProvider(provider)
+			return *providerInstance, nil
+		}
+	}
+	return tpmodels.TypeProvider{}, fmt.Errorf("the provider %s could not be found in the configuration", thirdPartyId)
+}
+
+func createProvider(input tpmodels.ProviderInput) *tpmodels.TypeProvider {
+	switch input.Config.ThirdPartyId {
+	case "active-directory":
+		return providers.ActiveDirectory(input)
+	case "apple":
+		return providers.Apple(input)
+	case "discord":
+		return providers.Discord(input)
+	case "facebook":
+		return providers.Facebook(input)
+	case "github":
+		return providers.Github(input)
+	case "google":
+		return providers.Google(input)
+	case "google-workspaces":
+		return providers.GoogleWorkspaces(input)
+	case "okta":
+		return providers.Okta(input)
+	case "linkedin":
+		return providers.Linkedin(input)
+	case "boxy-saml":
+		return providers.BoxySaml(input)
+	}
+
```

Comment:
You need to add to PR checklist that this list needs to be updated each time we add a new in built provider.

Reply:
It is there already in the base branch